### PR TITLE
Use v8 to test wasm binaries are valid in test suite binary format checks

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: scripts/test/shared.py [hunk 1]
```diff
@@ -389,6 +409,16 @@ def binary_format_check(wast, verify_final_result=True, wasm_as_args=['-g'],
...
 
+  # make sure it is a valid wasm, using a real wasm VM
+  if V8:
+    if os.path.basename(original_wast or wast) not in [
+      'atomics.wast',  # https://bugs.chromium.org/p/v8/issues/detail?id=9425
+      'simd.wast',  # https://bugs.chromium.org/p/v8/issues/detail?id=8460
+    ]:
+      cmd = [V8] + V8_OPTS + [in_binaryen('scripts', 'validation_shell.js'), '--', 'a.wasm']
+      print '      ', ' '.join(cmd)
+      subprocess.check_call(cmd, stdout=subprocess.PIPE)
```
Comment (aheejin) [inline scripts/test/shared.py]: Maybe print a warning message if v8 does not exist?

File: scripts/validation_shell.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Test a file is valid, by just loading it.
+
+// Shell integration.
+if (typeof console === 'undefined') {
+  console = { log: print };
+}
+var tempRet0;
```
Comment (aheejin) [inline scripts/validation_shell.js]: Where is this used?

File: scripts/validation_shell.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+var binary;
+if (typeof process === 'object' && typeof require === 'function' /* node.js detection */) {
+  var args = process.argv.slice(2);
+  binary = require('fs').readFileSync(args[0]);
+  if (!binary.buffer) binary = new Uint8Array(binary);
+} else {
+  var args;
+  if (typeof scriptArgs != 'undefined') {
```
Comment (aheejin) [inline scripts/validation_shell.js]: What is `scriptArgs`? (I don't know much about the JS part, sorry)

File: scripts/validation_shell.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  if (typeof readbuffer === 'function') {
+    binary = new Uint8Array(readbuffer(args[0]));
+  } else {
+    binary = read(args[0], 'binary');
+  }
+}
+
+// Test the wasm for validity. The imports aren't there (we can't use a proxy
+// because we don't know the types), but if we get to the imports, that means
+// the wasm is valid.
```
Comment (aheejin) [inline scripts/validation_shell.js]: Don't know much about the v8 validation process. Are imports validated after all other things?

File: scripts/validation_shell.js (hunk 3 above)
Comment (kripken) [reply scripts/validation_shell.js]: It's one of the ways various JS shells give us access to the commandline params. I think scriptArgs is for spidermonkey and arguments is for v8 (or maybe the opposite).

File: scripts/validation_shell.js (hunk 4 above)
Comment (kripken) [reply scripts/validation_shell.js]: Yes, it compiles first, then uses imports only when creating an instance (at least v8 and sm do - perhaps chakra does lazy validation/compilation?).

Thinking about this again now, I see the code can be much simpler - we can just compile and not even create an instance, so no imports are involved.

**Comment by kripken** [comment]
Thanks, that commit should address those comments.

Except for the note about other wasm files in test/ - good point, we should check them too. I'll do it as a followup.

**Comment by kripken** [comment]
Turns out it was just the wasm-dis checks - I was worried there were more. So I added those other wasm files in this PR.

Removed one old invalid file while doing so.

**Pull request merged**